### ISSUE TITLE
[Git] Timezone of author are not respected

### ISSUE THREAD

**Issue by andygrunwald** [MetricsGrimoire/CVSAnalY#63]
During some testing of pull requests of @linzhp i recognized that the time zone of commits are not respected.
The `git log` of input.tar.gz in test/ directory shows something like:

```
commit 456a68ee1407a77f3e804a30dff245bb6c6b872f
Merge: ce8e0b8 51a3b65
Author: Zhongpeng Lin (林中鹏) <lin.zhp@gmail.com>
Date:   Tue Feb 11 22:10:39 2014 -0800

    Merge branch 'lzp'

    Conflicts:
        aaa/otherthing

commit 51a3b654f252210572297f47597b31527c475fb8
Author: Zhongpeng Lin (林中鹏) <lin.zhp@gmail.com>
Date:   Tue Feb 11 22:09:26 2014 -0800

    modify aaa/otherthing

commit ce8e0b86a1e9877f42fe9453ede418519115f367
Author: Zhongpeng Lin (林中鹏) <lin.zhp@gmail.com>
Date:   Tue Feb 11 22:07:49 2014 -0800

    rename aaa/otherthing

commit 589bb080f059834829a2a5955bebfd7c2baa110a
Author: Eduardo Morais <companheiro.vermelho@gmail.com>
Date:   Tue Aug 14 15:04:01 2012 -0300

    Create "deeply" nested file
...
```

The `scmlog` table only stores `2014-02-11 22:10:39` (see last commit), without the timezone.

I my opinion, the timezone should be stored, too.
This information can be used for analysis how distributed the development of a repository is.
What do you think?

**Comment by dicortazar** [comment]
I totally agree with you, and for both: time of committer and time of author.

**Comment by dicortazar** [comment]
btw, the issue here is that MySQL does not store the TZ in the date field, so we should probably create another field similar to the one found in the Mailing List Stats "Messages" table.

**Comment by andygrunwald** [comment]
+1
How the implementation details looks like i didn`t thought about yet.
But maybe i will implement it. Thanks for your suggestion.
Of course you can implement this as well ;)

**Comment by jgbarah** [comment]
Maybe we could try to rise this a bit in our priority list. This issue has popped up several times: having TZ information would allow us to make TZ analysis for commits, which in some cases could be rather interesting.
And yes, using the MLStats approach is the best one, in my opinion.

**Comment by andygrunwald** [comment]
Can you provide a link to the mentioned solution?
I didnt know this

Am Donnerstag, 27. März 2014 schrieb Jesus M. Gonzalez-Barahona :

**Comment by jgbarah** [comment]
It is just the *_tz fields in the messages table, which stores the difference with UTC in seconds. See the schema at: https://github.com/MetricsGrimoire/MailingListStats/wiki/Database-Schema

For CVSAnalY, I guess we would need date_tz and author_date_tz, to be filled in with difference with UTC (eg, +3600 for CET, +7200 for CEST) when storing the corresponding datetime. Since the difference is available in the git log, as eg +0100 or +0200, it should be easy to produce those new two fields.

**Comment by andygrunwald** [comment]
At a first look (without a research) it looks logical :)

**Comment by sduenas** [comment]
Closed by #81
